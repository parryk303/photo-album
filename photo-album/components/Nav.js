import Link from 'next/link'
import Image from 'next/image'

const Nav = ({ albums, setAlbums, results, setResults }) => {
    const searchAlbums = (string) => {
        let result = [];
        albums.forEach(album => {
            const title = album.id.toString();
            if (title.includes(string)) {
                result.push(album);
            } else {
                return null;
            }
            setResults(result)
        });
        let photos = []
        photos.forEach(photo => {
            const title = photo.id.toString();
            if (title.includes(string)) {
                result.push(photo);
            } else {
                return null;
            }
            setPhotos(photos)
        });

    }

    return (
        <nav className='nav'>
            <Link href='/'>
                <div id='title'>
                    <Image src='/albumIcon.png' id='logo' width={150} height={150} />
                    <a className='brand'>Photo Album App</a>
                </div>
            </Link>
            <div className='ui search'>
                <div className='ui icon input'>
                    <input className='prompt' type='text' placeholder='Search albums...'
                        onChange= {e => {
                            searchAlbums( e.target.value.toLowerCase().trim());
                          }}/>
                    <i className='search icon'></i>
                </div>
                <div className='results'></div>
            </div>
        </nav>
    )
}

export default Nav;