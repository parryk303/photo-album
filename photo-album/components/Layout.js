import Head from 'next/head'
import Nav from './Nav'

const Layout = ({ albums, setAlbums, children, results, setResults }) => (
  <>
    <Head>
      <link
        async
        rel='stylesheet'
        href='https://cdn.jsdelivr.net/npm/semantic-ui@2/dist/semantic.min.css' />
      <title>Photo Album App</title>
      <link rel='icon' href='/favicon.ico' />
    </Head>
    <Nav
      albums={albums}
      setAlbums={setAlbums}
      results={results}
      setResults={setResults}/>
    {children}
  </>
)

export default Layout;