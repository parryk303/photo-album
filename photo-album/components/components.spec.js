/**
 * @jest-environment jsdom
 */
const { shallow, mount, render, configure } = require('enzyme')
import renderer from 'react-test-renderer'

import Footer from './Footer'
import Layout from './Layout'
import Nav from './Nav'

const sample = [
  {
    id: 1,
    title: "quidem molestiae enim",
    userId: 1
  },
  {
    id: 2,
    title: "sunt qui excepturi placeat culpa",
    userId: 1
  }
];

const sampleB = [
  {
    albumId: 1,
    id: 1,
    thumbnailUrl: "https://via.placeholder.com/150/92c952",
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952"
  },
  {
    albumId: 1,
    id: 2,
    thumbnailUrl: "https://via.placeholder.com/150/771796",
    title: "reprehenderit est deserunt velit ipsam",
    url: "https://via.placeholder.com/600/771796"
  }
];

describe('Footer', () => {
  it('renders Powered by text', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper.find('a').text()).toEqual('Powered by ');
  });
});

describe('Footer Snapshot', () => {
  it('renders with no errors', () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('Layout', () => {
  it('renders with albums and results props', () => {
    const data = sample;
    const wrapper = mount(<Layout albums={data} results={data} />);
    expect(wrapper.props().albums[0]).toEqual(data[0]);
    expect(wrapper.props().albums[1]).toEqual(data[1]);
    expect(wrapper.props().results[0]).toEqual(data[0]);
    expect(wrapper.props().results[1]).toEqual(data[1]);
  });
});

describe('Layout Snapshot', () => {
  it('renders with no errors', () => {
    const component = renderer.create(<Layout />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('Nav', () => {
  it('renders Photo Album App text', () => {
    const wrapper = mount(<Nav />);
    expect(wrapper.find('a').text()).toEqual('Photo Album App');
  });
});

describe('Nav', () => {
  it('renders with albums and results props', () => {
    const data = sample;
    const wrapper = mount(<Nav albums={data} results={data} />);
    expect(wrapper.props().albums[0]).toEqual(data[0]);
    expect(wrapper.props().albums[1]).toEqual(data[1]);
    expect(wrapper.props().results[0]).toEqual(data[0]);
    expect(wrapper.props().results[1]).toEqual(data[1]);
  });
});

describe('Nav Snapshot', () => {
  it('renders with no errors', () => {
    const component = renderer.create(<Nav />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})

