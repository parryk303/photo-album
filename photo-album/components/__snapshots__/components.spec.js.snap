// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot renders with no errors 1`] = `
<footer
  className="footer"
>
  <a
    href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
    rel="noopener noreferrer"
    target="_blank"
  >
    Powered by
     
    <span
      className="logo"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="Vercel Logo"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Vercel Logo"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fvercel.svg&w=256&q=75"
            srcSet="/_next/image?url=%2Fvercel.svg&w=96&q=75 1x, /_next/image?url=%2Fvercel.svg&w=256&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </span>
  </a>
</footer>
`;

exports[`Layout Snapshot renders with no errors 1`] = `
<nav
  className="nav"
>
  <div
    id="title"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        id="logo"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          id="logo"
          loading="lazy"
          src="/_next/image?url=%2FalbumIcon.png&w=384&q=75"
          srcSet="/_next/image?url=%2FalbumIcon.png&w=256&q=75 1x, /_next/image?url=%2FalbumIcon.png&w=384&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <a
      className="brand"
    >
      Photo Album App
    </a>
  </div>
  <div
    className="ui search"
  >
    <div
      className="ui icon input"
    >
      <input
        className="prompt"
        onChange={[Function]}
        placeholder="Search albums..."
        type="text"
      />
      <i
        className="search icon"
      />
    </div>
    <div
      className="results"
    />
  </div>
</nav>
`;

exports[`Nav Snapshot renders with no errors 1`] = `
<nav
  className="nav"
>
  <div
    id="title"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        id="logo"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          id="logo"
          loading="lazy"
          src="/_next/image?url=%2FalbumIcon.png&w=384&q=75"
          srcSet="/_next/image?url=%2FalbumIcon.png&w=256&q=75 1x, /_next/image?url=%2FalbumIcon.png&w=384&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <a
      className="brand"
    >
      Photo Album App
    </a>
  </div>
  <div
    className="ui search"
  >
    <div
      className="ui icon input"
    >
      <input
        className="prompt"
        onChange={[Function]}
        placeholder="Search albums..."
        type="text"
      />
      <i
        className="search icon"
      />
    </div>
    <div
      className="results"
    />
  </div>
</nav>
`;
